src/services/user/user-router.test.ts

```diff
@@ -97,54 +96,65 @@ beforeEach(async () => {
 
-        const jwtReturned = mockedGenerateJwtToken.mock.results[mockedGenerateJwtToken.mock.results.length - 1]!.value;
-
-        expect(JSON.parse(response.text)).toMatchObject({
+        // Verify response format
+        expect(responseBody).toMatchObject({
             userId: TESTER_USER.userId,
-            qrInfo: `hackillinois://user?userToken=${jwtReturned}`,
+            qrInfo: expect.stringMatching(/hackillinois:\/\/user\?qrId=[A-Za-z0-9+/=]+/),
         });
 
-        expect(mockedGenerateJwtToken).toBeCalledWith(
-            expect.objectContaining({
-                id: TESTER_USER.userId,
-            }),
-            false,
-            Config.QR_EXPIRY_TIME,
-        );
+        // Decrypt the QR code
+        const encryptedToken: string = responseBody.qrInfo.split("=")[1];
+        const decryptedData = decryptQR(encryptedToken);
+
+        // Verify decrypted data
+        expect(decryptedData.userId).toBe(TESTER_USER.userId);
+        expect(decryptedData.exp > creationTime).toBe(true);
     });
 });
 
 describe("GET /user/qr/:id/", () => {
-    it("gives a forbidden error for a non-staff user", async () => {
-        const response = await getAsAttendee(`/user/qr/${OTHER_USER.userId}/`).expect(StatusCode.ClientErrorForbidden);
-
-        expect(JSON.parse(response.text)).toHaveProperty("error", "Forbidden");
+    it("rejects non-staff users", async () => {
+        const response = await getAsAttendee(`/user/qr/${TESTER_USER.userId}/`).expect(StatusCode.ClientErrorForbidden);
+        expect(JSON.parse(response.text)).toMatchObject({
+            error: "Forbidden",
+        });
     });
 
-    it("works for a staff user", async () => {
-        const mockedGenerateJwtToken = mockGenerateJwtTokenWithWrapper();
+    it("generates QR code for specified user (staff)", async () => {
+        const response = await getAsStaff(`/user/qr/${TESTER_USER.userId}/`).expect(StatusCode.SuccessOK);
+        const responseBody = JSON.parse(response.text);
 
-        const response = await getAsStaff(`/user/qr/${OTHER_USER.userId}/`).expect(StatusCode.SuccessOK);
+        // Verify response
+        expect(responseBody).toMatchObject({
+            userId: TESTER_USER.userId,
+            qrInfo: expect.stringMatching(/hackillinois:\/\/user\?qrId=[A-Za-z0-9+/=]+/),
+        });
 
-        const jwtReturned = mockedGenerateJwtToken.mock.results[mockedGenerateJwtToken.mock.results.length - 1]!.value;
+        // Decrypt the QR code
+        const encryptedToken: string = responseBody.qrInfo.split("=")[1];
+        const decryptedData = decryptQR(encryptedToken);
 
-        expect(JSON.parse(response.text)).toMatchObject({
-            userId: OTHER_USER.userId,
-            qrInfo: `hackillinois://user?userToken=${jwtReturned}`,
-        });
+        // Verify decrypted data
+        expect(decryptedData.userId).toBe(TESTER_USER.userId);
+        expect(decryptedData.exp).toBeGreaterThan(Math.floor(Date.now() / 1000));
```

Comment:
Should be checking farther in the future - ideally half of expire time to verify it doesn't expire right away